tests/test_neighbors.py

```diff
@@ -47,3 +48,17 @@ def test_neighbors_key_added():
     assert np.allclose(
         adata.obsp["distances"].toarray(), adata.obsp[dists_key].toarray()
     )
+
+def test_bbknn():
+    adata = pbmc68k_reduced()
+    sce.pp.bbknn(adata, n_pcs=15,batch_key="phase",computation="pynndescent", metric="euclidean")
+    bbknn(adata, n_pcs=15,batch_key="phase",algorithm="brute", key_added="rapids")
+    counter = 0
+    for i in range(adata.shape[0]):
+        rsc_start =adata.obsp["rapids_distances"].indptr[i]
+        rsc_stop = adata.obsp["rapids_distances"].indptr[i+1]
+        b_start=adata.obsp["distances"].indptr[i]
+        b_stop=adata.obsp["distances"].indptr[i+1]
```

Comment:
you can replace all that with `for (rsc_start, rsc_stop), (b_start, b_stop) in zip(itertools.pairwise(...), itertools.pairwise(...)):`

Reply:
i dont know that this is more readable

Reply:
I would agree with phil here.  That's what these functions are for.

Reply:
I definitely recommend learning to use all of these standardlibs:
- functools
- itertools
- contextlib
- collections
- collections.abc